ostd/src/cpu/mod.rs

```diff
@@ -21,6 +21,70 @@ use crate::{
     arch::boot::smp::get_num_processors, task::DisabledPreemptGuard, trap::DisabledLocalIrqGuard,
 };
 
+/// The ID of a CPU in the system.
+///
+/// If converting from/to an integer, the integer must start from 0 and be less
+/// than the number of CPUs.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct CpuId(u32);
+
+/// An error indicating that the CPU does not exist.
+#[derive(Debug)]
+pub struct NonexistentCPU(pub usize);
+
+impl CpuId {
+    /// Returns the CPU ID of the bootstrap processor (BSP).
+    pub const fn bsp() -> Self {
+        CpuId(0)
+    }
+
+    /// Converts the CPU ID to an `usize`.
+    pub const fn as_usize(self) -> usize {
```

Comment:
Use `&self` seems to be enough.

Reply:
All conversions have a convention to consume the original value. And `CpuId` is `Copy` so I think it brings no inconvenience.

Reply:
Using `self` is cool here.